lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -1410,4 +1410,125 @@ libjit_dump_tensor(uint8_t *tensor, size_t *tensorDim, size_t numDimsTensor,
     break;
   }
 }
+
+static void find_min_max_f(float *tensor, size_t size, float &min, float &max) {
+  min = tensor[0];
+  max = tensor[0];
+
+  for (size_t i = 1; i < size; ++i) {
+    float tensorVal = tensor[i];
+    if (tensorVal < min)
+      min = tensorVal;
+
+    if (tensorVal > max)
+      max = tensorVal;
+  }
+}
+
+static int check_all_zeros(float *arrayToCheck, size_t size) {
```

Comment:
BTW, you can avoid loads/stores in the loop (corresponding to reads/writes of arguments `min` and `max` passed by reference) by keeping the running min & max in local variables (and only writing at the end) -- as in the `VERSION_LOCAL` example in https://godbolt.org/z/JuzDsI (compare with `VERSION_BASELINE`; cf. also `VERSION_MINMAX` using `std::minmax_element` if using std. lib. algorithms is okay).

Reply:
@ayermolo nvm, that's libjit which has different convention...

Reply:
@MattPD Thank you for your feedback. Is such a low level optimization really necessary? Reason I say this is that glow/llvm will inline this min_max function. It will then specialize libjit_quantization_profile. Looking at LLVM IR it will also do unroll and vectorization.

Reply:
@ayermolo Good point, not necessarily! If it's inlined into `libjit_quantization_profile` I wouldn't expect aliasing to be a problem (there are distinct local variables `minInput` and `maxInput` there, so no such aliasing is possible). One can even obtain the same generated code -- depending on the compiler & optimization level combination: https://godbolt.org/z/tURXvl vs https://godbolt.org/z/0kzSLR (the approach to branching can yield different results, though: https://godbolt.org/z/h7ykqj). Perhaps `std::minmax_element` could be the simplest choice, then (if using `<algorithm>` is okay for this).

Reply:
@rdzhabarov is using std libraries OK here?

Reply:
The libjit is first compiled into a bitcode and then this bitcode is included into the LLVM modules produced from a Glow IR of your NN model. Then it is compiled by means of LLVM's ORC JIT and executed. 

Thus any references to external libraries need to be resolved by the ORC JIT "linking" layer and in general we try to limit the amount of references to external functions if possible. 

That being said, references to `libc` functions are usually fine, as `libc` is typically present on any system. References to C++ libs could be more problematic, but on the other hand libc++ is also present on almost all systems these days. And if the std functions are just template functions that are expanded into a self-contained code, it should not be a problem. 

To check if it is OK to use std C++ functions, just try to use them in libjit and then try to run the NN model using the `-cpu` backend and see if it works or if you get some errors at runtime, because ORC JIT cannot find the referenced symbol. It is also important to check that the AOY functionality is not broken by such a change. To check this, try to create a bundle from your NN model and then link your app against the produced `*.o` file. It should not result in any linking errors.